internal/dao/patch.go

```diff
@@ -25,23 +30,31 @@ type Element struct {
 }
 
 // Build a json patch string to update PodSpec images
-func SetImageJsonPatch(images map[string]string) (string, error) {
-	elementOrders := make([]Element, 0)
-	containers := make([]Element, 0)
-	for key, value := range images {
-		elementOrders = append(elementOrders, Element{Name: key})
-		containers = append(containers, Element{Name: key, Image: value})
-	}
+func SetImageJsonPatch(spec v1.PodSpec) (string, error) {
 	jsonPatch := JsonPatch{
 		Spec: Spec{
 			Template: Template{
 				Spec: ImagesSpec{
-					SetElementOrders: elementOrders,
-					Containers:       containers,
+					SetElementOrderContainers:     extractElements(spec.Containers, false),
+					Containers:                    extractElements(spec.Containers, true),
+					SetElementOrderInitContainers: extractElements(spec.InitContainers, false),
+					InitContainers:                extractElements(spec.InitContainers, true),
 				},
 			},
 		},
 	}
 	bytes, err := json.Marshal(jsonPatch)
 	return string(bytes), err
 }
+
+func extractElements(containers []v1.Container, withImage bool) []Element {
+	elements := make([]Element, 0)
```

Comment:
Use var elements []Element. Better form in GO. Initializes a nil slice that can be appended to.